clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -750,6 +788,27 @@ public Node leastLoadedNode(long now) {
         if (algo == null) {
             throw new IllegalStateException("leastLoadedNodeAlgorithm cannot be null");
         }
+
+        if (this.metadataUpdater.isUpdateClusterMetadataDue(now) && !this.bootstrapServers.isEmpty()) {
```

Comment:
what happened if bootstrapServers.isEmpty and you get the wrong staled metadata from a different broker,  this should never happen in practice.  maybe instead of check bootstrapServers.isEmpty,  throw out some expectation if bootstrapServers.isEmpty?

Reply:
Discussed offline, if bootstrapServer.isEmpty is true, parseAndValidateAddresses will throw a ConfigException already, no need to add more exception throw, so just remove the empty check.